### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by anes66** [friped/fdev_weapons#2]
Hello thank you for sharing that script with us i have 1 question and 1 problem

i made a weapon addon and i named it  `    {weapon = 'WEAPON_GLOCK', item = 'glock', ammoItem = 'ammo_pd'},` and evrything is good when i have the 2 items i can shoot normaly but here is the problem when i have like `ammo_pistol` it uses this ammo to but i dont lose it in inventory so bassicly i have infinity ammo i just need to drop it and then take it again

I hope you understand what i mean if you need more info let me know 

So bassicly what i asking is can you make so for example
`    {weapon = 'WEAPON_GLOCK', item = 'glock', ammoItem = 'ammo_pd'},` and can not use ammo_pistol what is bassicly the same

**Comment by friped** [comment]
Well, this is a issue with the game as i understand this as the block you’ve added into game uses the same ammo as the normal pistol in the game and the script checks for the ammo type. I don’t know anything about adding custom weapons into the game but can you add custom ammotypes for the weapon this would be a great solution for your problem but as I wrote I don’t know how custom weapons work

**Comment by anes66** [comment]
> Well, this is a issue with the game as i understand this as the block you’ve added into game uses the same ammo as the normal pistol in the game and the script checks for the ammo type. I don’t know anything about adding custom weapons into the game but can you add custom ammotypes for the weapon this would be a great solution for your problem but as I wrote I don’t know how custom weapons work

Here you can make with this tool custom weapons its really easy and it takes 5min max to make a addon maybe that can help you it create evrything  what you need for fivem
  [tool for weapon](https://www.gta5-mods.com/tools/vweaponstoolki-addon-weapon-generator)

I have 1 more idea maybe if you can add like clip for every weapon

there must be a way to make it possible to make that xD

Or maybe to make a check if he has this item or different ammo type and if the name is not same it unload instant

**Comment by friped** [comment]
That would be extremely bad for the script as it needs to work more for the clients and is made at a way to be good for the game client

**Comment by anes66** [comment]
I will see what i can do ofc if i have from you the permision to edit it and test some stuff if i can maybe limit it or change so that weapons use only that item what i want


Im currently think about so
To create ammo item and if that is the right item for this weapon it will load the weapon with pistol ammo

bassicly like bassicneed
You cant eat burger or pizza and restore water or be high like on drugs xD

**Comment by friped** [comment]
Yea, change whatever you want as long as you don’t release it as a new code without creeds, I want it to work as much and greatly for everyone so if you get it to work just as you want and optimal you are welcome to post the solution here :)

**Comment by anes66** [comment]
> Yea, change whatever you want as long as you don’t release it as a new code without creeds, I want it to work as much and greatly for everyone so if you get it to work just as you want and optimal you are welcome to post the solution here :)

Ofc no problem and we will see

play a little bit with te tool if you didnt see in the second answer maybe you will have better idea as me or more luck xD

**Comment by friped** [comment]
Yes, I've played around a fair bit with custom weapons ingame now and found that you'll need to add a custom ammotype in a weaponmeta and add that to your weapon in the meta aswell for how it probably is now is that it runs under ordanary pistol

**Comment by anes66** [comment]
> 

a ok you mean under `<AmmoInfo ref="AMMO_PISTOL"/>` did you try AMMO_GLOCK or somthing like that.
Im not sure if i tested it but i think if you change it in weapons meta he still will use AMMO_PISTOL and AMMO_GLOCK

**Comment by friped** [comment]
Yes you change that to the ammo tyde you want but you also need to add it as a ammotype with the other ones there are full weapon.meta in the cfx forum with all original data where you simply can add it

**Comment by anes66** [comment]
> 

wait wait wait the tool what i give to you it create a custom weapons.meta

Here you can see did you try it with that?  how i remember it take the ammo_glock but try again with ammo_pistol i think it still use it no matter what i write there

**Comment by friped** [comment]
Ive tried to change a normal pistol in game and was needed to add the ammotype also before the weapon took the correct ammotype

**Comment by anes66** [comment]
Ok thats correct i made also like a item ammo_glock and for the pistol Glock i added ammo type to be used is ammo_glock  but maybe you found another way or i made some mistake i dont know :P but when i tested it i still can used ammo_pistol and ammo_glock 

or wait test it soo give to you Pistol with pistol ammo and give glock with glock ammo  then you will see i think it will use both ammo types

**Comment by JackDaniels1994** [comment]
have the same issue, have to drop weapon to use again, this script does not support it, use weaponSync, and its fine 🗡️
